### PR TITLE
fix(ACL): sanitize ACL search parameters 

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description
This PR fix an issue where ACL search parameters was vulnerable to injections.

**Fixes** # MON-5925 MON-5926 MON-5927

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 2.8.x

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/options/accessLists/actionsACL/listsActionsAccess.php [hunk 1]
```diff
@@ -36,20 +36,27 @@
 if (!isset($centreon)) {
     exit();
 }
-    
+
 include("./include/common/autoNumLimit.php");
+$num = (int) $num;
+$limit = (int) $limit;
```
Comment (sc979) [inline www/include/options/accessLists/actionsACL/listsActionsAccess.php:42]: please check that limit > 0. If not set a default value

File: www/include/options/accessLists/groupsACL/listGroupConfig.php [hunk 2]
```diff
@@ -38,18 +38,25 @@
 }
 
 include("./include/common/autoNumLimit.php");
+$num = (int) $num;
+$limit = (int) $limit;
```
Comment (sc979) [inline www/include/options/accessLists/groupsACL/listGroupConfig.php:42]: kindly check for negative num and limit

File: www/include/options/accessLists/resourcesACL/listsResourcesAccess.php [hunk 3]
```diff
@@ -38,18 +38,26 @@
 }
 
 include("./include/common/autoNumLimit.php");
+$num = (int) $num;
+$limit = (int) $limit;
```
Comment (sc979) [inline www/include/options/accessLists/resourcesACL/listsResourcesAccess.php:42]: same

File: www/include/options/accessLists/actionsACL/listsActionsAccess.php (hunk 1 above)
Comment (jeremyjaouen) [reply www/include/options/accessLists/actionsACL/listsActionsAccess.php:42]: all the checks are already done in autoNumLimit.php

File: www/include/options/accessLists/groupsACL/listGroupConfig.php (hunk 2 above)
Comment (jeremyjaouen) [reply www/include/options/accessLists/groupsACL/listGroupConfig.php:42]: all the checks are already done in autoNumLimit.php

File: www/include/options/accessLists/resourcesACL/listsResourcesAccess.php (hunk 3 above)
Comment (jeremyjaouen) [reply www/include/options/accessLists/resourcesACL/listsResourcesAccess.php:42]: all the checks are already done in autoNumLimit.php

**Review by sc979 — APPROVED** [review]

**Review by sc979 — CHANGES_REQUESTED** [review]
one PR, one scope
The jenkinsfile modifications should be done in another PR

**Review by sc979 — APPROVED** [review]

File: www/include/options/accessLists/actionsACL/listsActionsAccess.php [hunk 4]
```diff
@@ -1,55 +1,63 @@
...
 if (isset($_POST['searchACLA']) && $_POST['searchACLA']) {
-    $search = $_POST['searchACLA'];
-    $searchStr = "WHERE (acl_action_name LIKE '%".htmlentities($search, ENT_QUOTES, "UTF-8")
-        . "%' OR acl_action_description LIKE '".htmlentities($search, ENT_QUOTES, "UTF-8")."')";
+    $search = filter_var($_POST['searchACLA'], FILTER_SANITIZE_STRING);
+    $searchStr = "WHERE (acl_action_name LIKE ? OR acl_action_description LIKE ?)";
+}
+$statement = $pearDB->prepare("SELECT COUNT(*) FROM acl_actions " . $searchStr);
+if ($search) {
+    $result = $pearDB->execute($statement, array('%' . $search . '%', '%' . $search . '%'));
```
Comment (callapa) [inline www/include/options/accessLists/actionsACL/listsActionsAccess.php:53]:
```suggestion
    $statement = $pearDB->prepare("SELECT COUNT(*) FROM acl_actions WHERE (acl_action_name LIKE ? OR acl_action_description LIKE ?");
    $result = $pearDB->execute($statement, array('%' . $search . '%', '%' . $search . '%'));
```

File: www/include/options/accessLists/actionsACL/listsActionsAccess.php [hunk 5]
```diff
@@ -1,55 +1,63 @@
...
 $searchStr = "";
 $search = '';
 if (isset($_POST['searchACLA']) && $_POST['searchACLA']) {
-    $search = $_POST['searchACLA'];
-    $searchStr = "WHERE (acl_action_name LIKE '%".htmlentities($search, ENT_QUOTES, "UTF-8")
-        . "%' OR acl_action_description LIKE '".htmlentities($search, ENT_QUOTES, "UTF-8")."')";
+    $search = filter_var($_POST['searchACLA'], FILTER_SANITIZE_STRING);
+    $searchStr = "WHERE (acl_action_name LIKE ? OR acl_action_description LIKE ?)";
+}
+$statement = $pearDB->prepare("SELECT COUNT(*) FROM acl_actions " . $searchStr);
```
Comment (callapa) [inline www/include/options/accessLists/actionsACL/listsActionsAccess.php:51]:
```suggestion
```

File: www/include/options/accessLists/actionsACL/listsActionsAccess.php [hunk 6]
```diff
@@ -1,55 +1,63 @@
...
+$limit = (int) $limit;
 
 $searchStr = "";
 $search = '';
 if (isset($_POST['searchACLA']) && $_POST['searchACLA']) {
-    $search = $_POST['searchACLA'];
-    $searchStr = "WHERE (acl_action_name LIKE '%".htmlentities($search, ENT_QUOTES, "UTF-8")
-        . "%' OR acl_action_description LIKE '".htmlentities($search, ENT_QUOTES, "UTF-8")."')";
+    $search = filter_var($_POST['searchACLA'], FILTER_SANITIZE_STRING);
+    $searchStr = "WHERE (acl_action_name LIKE ? OR acl_action_description LIKE ?)";
```
Comment (callapa) [inline www/include/options/accessLists/actionsACL/listsActionsAccess.php:49]:
```suggestion
```

File: www/include/options/accessLists/actionsACL/listsActionsAccess.php [hunk 7]
```diff
@@ -1,55 +1,63 @@
...
+}
+$statement = $pearDB->prepare("SELECT COUNT(*) FROM acl_actions " . $searchStr);
+if ($search) {
+    $result = $pearDB->execute($statement, array('%' . $search . '%', '%' . $search . '%'));
+} else {
+    $result = $pearDB->execute($statement, array());
```
Comment (callapa) [inline www/include/options/accessLists/actionsACL/listsActionsAccess.php:55]:
```suggestion
    $statement = $pearDB->query("SELECT COUNT(*) FROM acl_actions");
    $result = $statement->execute();
```

**Review by adr-mo — APPROVED** [review]

**Pull request merged**